Classes/General.cs

```diff
@@ -89,7 +90,37 @@ public static class General {
                             try {
                                 if (aborting) return;
                                 response = (HttpWebResponse)request.EndGetResponse(requestResultParam);
-                                responseStream = new ThrottledStream(response.GetResponseStream(), Settings.MaximumBytesPerSecond ?? ThrottledStream.Infinite);
+                                if (response.ContentType == ContentType.TextHtml) {
+                                    MemoryStream memoryStream = new MemoryStream();
+                                    CopyStream(response.GetResponseStream(), memoryStream);
+                                    memoryStream.Position = 0;
+                                    byte[] redirectPageBytes = memoryStream.ToArray();
+                                    Encoding pageEncoding = DetectHTMLEncoding(redirectPageBytes, response.ContentType);
+                                    metaRedirectHtml = pageEncoding.GetString(redirectPageBytes);
+                                    memoryStream.Position = 0;
+                                    responseStream = memoryStream;
+                                    if (IsMetaRefresh(metaRedirectHtml)) {
+                                        HttpWebRequest redirectionRequest = null;
+                                        redirectionRequest = (HttpWebRequest)WebRequest.Create(GetRefreshUrl(metaRedirectHtml));
```

Comment:
I think we should move the creation of the request object to a reusable function since we now have duplicated code.

Reply:
Do you mean just moving the request into a function for checking if certain request headers should be set up?